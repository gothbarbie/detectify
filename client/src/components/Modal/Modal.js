import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const ModalBackdrop = styled.div`
  position: fixed;
  display: flex;
  align-items: center;
  justify-content: center;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.7);
`

const ModalWrapper = styled.section`
  background-color: #fff;
  border-radius: 4px;
  padding: 3rem 2rem;
`

const Modal = ({ show, children }) => {
  return show ? (
    <ModalBackdrop>
      <ModalWrapper>{children}</ModalWrapper>
    </ModalBackdrop>
  ) : null
}

Modal.propTypes = {
  children: PropTypes.node.isRequired,
  show: PropTypes.bool.isRequired,
}

export default Modal
