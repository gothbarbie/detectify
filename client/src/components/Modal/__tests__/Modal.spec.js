import React from 'react'
import { shallow } from 'enzyme'

import Modal from '../Modal'

describe('Components/Modal', () => {
  it('renders', () => {
    const wrapper = shallow(
      <Modal show={true}>
        <div>Content</div>
      </Modal>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
