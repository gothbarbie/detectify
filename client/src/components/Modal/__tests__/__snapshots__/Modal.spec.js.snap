// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Modal renders 1`] = `
<div>
  <div>
    Content
  </div>
</div>
`;
